src/video.cpp

```diff
@@ -5,861 +5,908 @@
-			switch (mode >> 6) {
-				case 0: // Fully opaque pixels.
-					mode &= 0x3F;
-					if (shift == GS_SEMI_TRANSPARENT) {
-						src += 3 * mode;
-						for (; mode > 0; mode--) {
-							if (is_in_drawing_region()) {
-								uint32 ndest = BlendPixels(255, 255, 255, *src_base, OPACITY_SEMI_TRANSPARENT);
-								BlitPixel(cr, src_base, xpos, ypos, numx, numy, spr->width, spr->height, ndest);
-							}
-							xpos++;
-							src_base++;
-						}
-					} else {
-						for (; mode > 0; mode--) {
-							if (is_in_drawing_region()) {
-								uint32 colour = MakeRGBA(sf(src[0]), sf(src[1]), sf(src[2]), OPAQUE);
-								BlitPixel(cr, src_base, xpos, ypos, numx, numy, spr->width, spr->height, colour);
-							}
-							xpos++;
-							src_base++;
-							src += 3;
-						}
-					}
-					break;
-
-				case 1: { // Partial opaque pixels.
-					uint8 opacity = *src++;
-					if (shift == GS_SEMI_TRANSPARENT && opacity > OPACITY_SEMI_TRANSPARENT) opacity = OPACITY_SEMI_TRANSPARENT;
-					mode &= 0x3F;
-					for (; mode > 0; mode--) {
-						if (is_in_drawing_region()) {
-							uint32 ndest = BlendPixels(sf(src[0]), sf(src[1]), sf(src[2]), *src_base, opacity);
-							BlitPixel(cr, src_base, xpos, ypos, numx, numy, spr->width, spr->height, ndest);
-						}
-						xpos++;
-						src_base++;
-						src += 3;
-					}
-					break;
-				}
-				case 2: // Fully transparent pixels.
-					xpos += mode & 0x3F;
-					src_base += mode & 0x3F;
-					break;
-
-				case 3: { // Recoloured pixels.
-					uint8 layer = *src++;
-					const uint32 *table = recolour.GetRecolourTable(layer - 1);
-					uint8 opacity = *src++;
-					if (shift == GS_SEMI_TRANSPARENT && opacity > OPACITY_SEMI_TRANSPARENT) opacity = OPACITY_SEMI_TRANSPARENT;
-					mode &= 0x3F;
-					for (; mode > 0; mode--) {
-						uint32 colour = table[*src++];
-						if (is_in_drawing_region()) {
-							colour = BlendPixels(sf(GetR(colour)), sf(GetG(colour)), sf(GetB(colour)), *src_base, opacity);
-							BlitPixel(cr, src_base, xpos, ypos, numx, numy, spr->width, spr->height, colour);
-						}
-						xpos++;
-						src_base++;
-					}
-					break;
-				}
-			}
-		}
-		line_base += cr.pitch;
-		ypos++;
-		src += 2; // Skip the length word.
+void VideoSystem::CoordsToGL(float *x, float *y) const {
+	float w, h;
+	if (this->clip.empty()) {
+		w = this->width;
+		h = this->height;
+	} else {
+		w = this->clip.back().width;
+		h = this->clip.back().height;
+		*x -= this->clip.back().base.x;
+		*y -= this->clip.back().base.y;
 	}
+
+	*x = 2.0f * *x / w - 1.0f;
+	*y = 1.0f - 2.0f * *y / h;
 }
 
 /**
- * Blit pixels from the \a spr relative to \a img_base into the area.
- * @param pt Base coordinates of the sprite data.
- * @param spr The sprite to blit.
- * @param numx Number of sprites to draw in horizontal direction.
- * @param numy Number of sprites to draw in vertical direction.
- * @param recolour Sprite recolouring definition.
- * @param shift Gradient shift.
+ * Load an OpenGL shader.
```

Comment:
It loads two shaders, maybe this is not about loading shaders but about configuring shaders for OpenGL?

(ie a higher level aim)